x-pack/plugins/observability_solution/observability_ai_assistant/server/plugin.ts

```diff
@@ -112,13 +114,16 @@ export class ObservabilityAIAssistantPlugin
 
     // Using once to make sure the same model ID is used during service init and Knowledge base setup
     const getModelId = once(async () => {
+      const configModelId = this.config.modelId;
       const defaultModelId = '.elser_model_2';
       const [_, pluginsStart] = await core.getStartServices();
       const license = await firstValueFrom(pluginsStart.licensing.license$);
-
       if (!license.hasAtLeast('enterprise')) {
         return defaultModelId;
       }
+      if (configModelId) {
```

Comment:
I put it below just to make sure someone without an enterprise license could not try to use it, but not sure it matters.  And because we are only, at the moment, testing the enterprise functionality using the assistant where its allowed.  Though, I'm not sure why we need to `doInit` to start with and create the ES assets if not enterprise.

Reply:
I think we await the license check here to make sure the license check is completed. We don't really care whether it is valid, from the looks of it - why even return a model id if the license is not valid? So I think we should move the license check out of this code, and throw an invalid license error further down. WDYT?

Reply:
It sounds like you and Søren had a similar conversation https://github.com/elastic/kibana/pull/181700/files#r1579341084

It seems like we were catching the invalid license error that ML was throwing but we didn't want to log it.  So we check if its enterprise and just return the "default model id" even though its no use.  I was also thinking we should not call doInit or stop getModelId from being called further up if they dont have the enterprise license.

I moved the config before license and added back in a comment that was removed that I think was helpful.